File: day1.js
const fs = require("node:fs");

function calcDistance (arrayOne, arrayTwo) {

    const group1Sorted = arrayOne.sort();
    const group2Sorted = arrayTwo.sort();
    
    const distances = group1Sorted.map((locID, i) => Math.abs( locID - group2Sorted[i]));
  
    return distances.reduce((totalDist, dist) => totalDist + dist);
}

fs.readFile("input.txt", "utf8", (err, data) => {
    if (err) {
        console.error(err);
        return;
    } 
    
    const lines = data.split('\n');

    const group1 = [];
    const group2 = [];

    for (line of lines) {
        group1.push(Number(line.split(/\s/).filter(n => n)[0]));
        group2.push(Number(line.split(/\s/).filter(n => n)[1]));
    }

    console.log("Total distance between the two lists:", calcDistance(group1, group2));
});

File: day1.py
#!/usr/bin/python3

group1 = []
group2 = []

with open("input.txt") as file: 
    for line in file:
        group1.append(int(line.split()[0]))
        group2.append(int(line.split()[1]))

group1.sort()
group2.sort()

total_dist = 0

for i in range(0,len(group1)):
    dist = abs( group1[i] - group2[i] )
    total_dist += dist
    #print(i, group1[i], group2[i], dist, total_dist)

print("Total distance between the two lists:", total_dist)
